src/NLog/Config/LoggingRuleLevelFilter/DynamicLogLevelFilter.cs

```diff
@@ -0,0 +1,132 @@
+﻿// 
+// Copyright (c) 2004-2019 Jaroslaw Kowalski <jaak@jkowalski.net>, Kim Christensen, Julian Verdurmen
+// 
+// All rights reserved.
+// 
+// Redistribution and use in source and binary forms, with or without 
+// modification, are permitted provided that the following conditions 
+// are met:
+// 
+// * Redistributions of source code must retain the above copyright notice, 
+//   this list of conditions and the following disclaimer. 
+// 
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution. 
+// 
+// * Neither the name of Jaroslaw Kowalski nor the names of its 
+//   contributors may be used to endorse or promote products derived from this
+//   software without specific prior written permission. 
+// 
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE 
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE 
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE 
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR 
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) 
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF 
+// THE POSSIBILITY OF SUCH DAMAGE.
+// 
+
+namespace NLog.Config
+{
+    using System;
+    using System.Collections.Generic;
+    using NLog.Common;
+    using NLog.Internal;
+    using NLog.Layouts;
+
+    /// <summary>
+    /// Dynamic filtering with a positive list of enabled levels
+    /// </summary>
+    internal class DynamicLogLevelFilter : ILoggingRuleLevelFilter
+    {
+        private static readonly char[] _levelFilerSplitter = { ',' };
+        private readonly LoggingRule _loggingRule;
+        private readonly SimpleLayout _levelFilter;
+        private KeyValuePair<string, bool[]> _activeFilter;
+
+        public bool[] LogLevels => GenerateLogLevels();
+
+        public DynamicLogLevelFilter(LoggingRule loggingRule, SimpleLayout levelFilter)
+        {
+            _loggingRule = loggingRule;
+            _levelFilter = levelFilter;
+            _activeFilter = new KeyValuePair<string, bool[]>(string.Empty, LoggingRuleLevelFilter.Off.LogLevels);
+        }
+
+        public LoggingRuleLevelFilter GetSimpleFilterForUpdate()
+        {
+            return new LoggingRuleLevelFilter(LogLevels);
+        }
+
+        private bool[] GenerateLogLevels()
```

Comment:
how critical is the performance of this method?

Reply:
What are you guiding towards?

Reply:
refactoring

Reply:
by e.g replacing bool[] with a specialized class

Reply:
I can make refactoring for you without anyone being hurt.

Reply:
Refactoring complete

Reply:
If I would create a class that encapsulates bool[], do you think that would be a bad thing for the performance?

Reply:
No hidding the bool[] array away and making methods instead on `ILoggingRuleLevelFilter` should not hurt performance.

Reply:
But please make that in a new PR instead of adding more into this one.

Reply:
Ok, before of after merge this PR?

Reply:
After merge is easier for me :)